wagtail/admin/templates/wagtailadmin/notifications/task_state_approved.html

```diff
@@ -0,0 +1,8 @@
+{% extends 'wagtailadmin/notifications/base.html' %}
+{% load i18n %}
+
+{% block content %}
+    <p>{% blocktrans with title=page.get_admin_display_title|safe task=task.name %}The page "{{ title }}" has been approved in moderation stage "{{ task }}".{% endblocktrans %}</p>
+    <p>{% trans "You can preview the page here:" %} <a href="{{ settings.BASE_URL }}{% url 'wagtailadmin_pages:preview_on_edit' page.id %}">{{ settings.BASE_URL }}{% url 'wagtailadmin_pages:preview_on_edit' page.id %}</a><br/></p>
```

Comment:
I don't think `preview_on_edit` is the right URL to use here - usually that view is called with a POST (to stash the data from the edit form into the session) followed by a GET (to retrieve the stashed data and render a preview based on it), so the GET on its own will fail (or serve up some arbitrary revision of the page if they happen to have been editing it earlier in their session). I think `preview_for_moderation` (which takes a revision ID) is the one we want, unless that's also been refactored in the workflow work...

Reply:
I've left `preview_for_moderation` as is for now, but I think we may well need another view, as while TaskStates are associated with revisions, when approving or rejecting, you're always intended to approve/reject the latest - there shouldn't be more than one revision in moderation at once. (that said, the logic to create a new TaskState tracking the new revision if a page is edited isn't in yet)

Reply:
Re "there shouldn't be more than one revision in moderation at once" - the existing moderation mechanism works on this basis too, but we use the revision ID to avoid race conditions - if an author edits the page in between the moderator opening the preview and approving it, they could potentially sneak in unwanted edits. Is workflow approval vulnerable to this?

Reply:
I've added a way to pass TaskStates through to the workflow action view to avoid this, and added a new moderation preview view to avoid altering the other too much. (It does not have the approve/reject buttons yet on the preview, but I think that can be another PR?)